wallet-new/src/Cardano/Wallet/Kernel/CoinSelection/Generic/Fees.hs

```diff
@@ -150,14 +144,22 @@ feeFromChange totalFee =
 -- as unchanged as possible
 reduceChangeOutputs :: forall dom. CoinSelDom dom
                     => Fee dom -> [Value dom] -> ([Value dom], Fee dom)
-reduceChangeOutputs totalFee [] = ([], totalFee)
 reduceChangeOutputs totalFee cs =
-      bimap identity unsafeFeeSum
-    . unzip
-    . map go
-    . divvyFee identity totalFee
-    $ cs
+    case cs' of
+        [] ->
+            (cs0, totalFee)
+
+        _  ->
+            bimap (++ cs0) unsafeFeeSum
+            . unzip
+            . map go
+            . divvyFee identity totalFee
+            $ cs'
   where
+    -- divvyFee only accepts non-zero output values as it doesn't make sense
+    -- to apply fees to those.
+    (cs0, cs') = partition (== valueZero) cs
```

Comment:
Do we need this `partition`, at all? With @edsko we initially thought a simple `filter` for `(> valueZero)` might have been enough. What are we missing?

Reply:
As far as I understand, we still want to return those zero value as part of the change, hence the change from 

```hs
 bimap identity unsafeFeeSum
```

to 

```hs
 bimap (++ cs0) unsafeFeeSum
```

We just don't run the `divvyFee` repartition function on them... I mean, we would filter out the zero value at this stage, but I'd expect this function to return the same number of inputs I gave it, no ?